entry.go

```diff
@@ -82,7 +82,9 @@ func (entry *Entry) WithFields(fields Fields) *Entry {
 	for k, v := range fields {
 		data[k] = v
 	}
-	return &Entry{Logger: entry.Logger, Data: data}
+	newEntry := *entry
```

Comment:
This shallow copy will inadvertently share the buffer. I'm not sure if that is intended.